src/query/api/v1/handler/prometheus/native/read_common.go

```diff
@@ -224,27 +228,18 @@ func insertSortedBlock(
 			"the block, wanted: %d, found: %d", seriesCount, blockSeriesCount)
 	}
 
-	blockStepIter, err := b.StepIter()
-	if err != nil {
-		return nil, err
-	}
-
-	blockStepCount := blockStepIter.StepCount()
-	if stepCount != blockStepCount {
-		return nil, fmt.Errorf("mismatch in number of steps for the"+
-			"block, wanted: %d, found: %d", stepCount, blockStepCount)
-	}
-
 	// Binary search to keep the start times sorted
 	index := sort.Search(len(blockList), func(i int) bool {
-		return blockList[i].meta.Bounds.Start.Before(blockMeta.Bounds.Start)
+		return blockList[i].meta.Bounds.Start.After(blockMeta.Bounds.Start)
```

Comment:
Why change this?

Reply:
Because it was wrong; it sorted in the wrong order